apps/client/src/components/form/ChallengesForm.tsx

```diff
@@ -0,0 +1,89 @@
+'use client';
+
+import { z } from 'zod';
+import { useForm, zodResolver } from '@mantine/form';
+import { Button, Flex, NumberInput, Paper, Textarea, TextInput } from '@mantine/core';
+
+const schema = z.object({
+  challengeScore: z.number().min(1, {
+    message: 'Please enter a valid score'
+  }),
+  challengeTitle: z.string().min(2, {
+    message: 'Please enter a valid title'
+  }),
+  description: z.string().min(10, {
+    message: 'Please enter a valid description that is clear'
+  })
+});
+
+export interface Challenge {
+  challengeScore: number;
+  challengeTitle: string;
+  description: string;
+}
+
+const ChallengesForm = () => {
+  const form = useForm<Challenge>({
+    initialValues: {
+      challengeScore: NaN,
+      challengeTitle: '',
+      description: ''
+    },
+    validate: zodResolver(schema),
+    validateInputOnBlur: true,
+    name: 'challenge'
+  });
+
+  const handleSubmit = async (values: Challenge) => {
+    form.validate();
+    // TODO: handle submit
+    try {
+      console.log('submit', values);
+      form.reset();
+    } catch (error) {
+      // Error handling
+    }
+  };
+
+  return (
+    <Paper withBorder radius={'8px'} shadow="md" p="1.5rem">
+      <form onSubmit={form.onSubmit(handleSubmit)} onReset={form.reset}>
+        <Flex direction="column" gap={'1rem'}>
+          <Flex direction="row" justify="space-between" gap="1em">
+            <NumberInput
+              radius={'6px'}
+              hideControls
+              w={'100%'}
+              label="Challenge Score"
+              placeholder="Enter challenge score"
+              {...form.getInputProps('challengeScore')}
+            />
+            <TextInput
```

Comment:
lets have the text input take 3/4th of the space while score takes 1/4th 

we can also try 4/5 and 1/5 split if that looks better as title would take way more space than just points .

Reply:
Done